src/box/lua/config/instance_config.lua

```diff
@@ -98,127 +73,24 @@ local function enterprise_edition_apply_default_if(_data, _w)
     return tarantool.package == 'Tarantool Enterprise'
 end
 
--- Generate a function that calls two given functions in a row.
---
--- If one of the arguments is nil, return the other one.
---
--- If both arguments are nil, returns nil.
-local function chain2(f1, f2)
-    if f1 == nil then
-        return f2
-    end
-    if f2 == nil then
-        return f1
-    end
-    return function(...)
-        f1(...)
-        f2(...)
-    end
-end
-
 -- Available only in Tarantool Enterprise Edition.
 local function enterprise_edition(schema_node)
     schema_node.enterprise_edition = true
+    schema_node.validate = enterprise_edition_validate
 
     -- Perform a domain specific validation first and only then
     -- check, whether the option is to be used with Tarantool
     -- Enterprise Edition.
     --
     -- This order is consistent with data type validation, which
     -- is also performed before the EE check.
-    schema_node.validate = chain2(
-        schema_node.validate,
-        enterprise_edition_validate)
-    schema_node.apply_default_if = chain2(
+    schema_node.apply_default_if = funcutils.chain2(
```

Comment:
I like the idea of splitting this patchset to three commits, but if we do, we should move the function (and fix the caller code) in the appropriate commits (the first one for `chain2` and the second one for `uri_is_suitable_to_connect`).

Reply:
> but if we do, we should move the function (and fix the caller code) in the appropriate commits

Yes, it should have been done that way. I've fixed it.